sqld/src/http/mod.rs

```diff
@@ -247,16 +250,48 @@ pub async fn run_http<D: Database>(
     upgrade_tx: mpsc::Sender<hrana::ws::Upgrade>,
     hrana_http_srv: Arc<hrana::http::Server<D>>,
     enable_console: bool,
-    idle_shutdown_layer: Option<IdleShutdownLayer>,
+    _idle_shutdown_layer: Option<IdleShutdownLayer>,
     stats: Stats,
 ) -> anyhow::Result<()> {
+    let state = AppState {
+        auth,
+        db_factory,
+        upgrade_tx,
+        hrana_http_srv,
+        enable_console,
+        stats,
+    };
+
     tracing::info!("listening for HTTP requests on {addr}");
 
     fn trace_request<B>(req: &Request<B>, _span: &Span) {
         tracing::debug!("got request: {} {}", req.method(), req.uri());
     }
-    let service = ServiceBuilder::new()
-        .option_layer(idle_shutdown_layer)
+
+    let app = Router::new()
+        .route("/", post(handle_query))
+        .route("/", get(handle_upgrade))
+        .route("/version", get(handle_version))
+        .route("/console", get(show_console))
+        .route("/health", get(handle_health))
+        .route("/v1/stats", get(stats::handle_stats))
+        .route("/v1/", get(hrana_over_http_1::handle_index))
+        .route("/v1/execute", post(hrana_over_http_1::handle_execute))
+        .route("/v1/batch", post(hrana_over_http_1::handle_batch))
+        .route("/v2", get(crate::hrana::http::handle_index))
+        .route("/v2/pipeline", post(handle_hrana_v2))
+        .fallback(handle_fallback)
+        .with_state(state);
+
+    let layered_app = app
+        // TODO: error mismatch needs to be fixed
+        // why? the option layer from tower merges the errors by boxing them
+        // and this breaks how axum uses infalliable. Though the option layer
```

Comment:
just curious -- can you elaborate? It's the second time you mention axum stuff being unboxable, and I wonder why it's the case. Is it documented somewhere so I can educate myself on it?

Reply:
Basically the shape the tower itself does things is by merging all errors into Box error, but axum instead uses infallible and this can cause trait mismatches. It turns out though that there is https://docs.rs/axum-extra/latest/axum_extra/middleware/fn.option_layer.html which we can use for this exact use case.

Reply:
This has been fixed in https://github.com/libsql/sqld/pull/572/commits/a08bf6c486eb674b6befc0d62038595bba24e49b